src/concurrent_handshake.rs

```diff
@@ -0,0 +1,64 @@
+use super::AsyncTls;
+use pin_project_lite::pin_project;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+use tokio::io::{AsyncRead, AsyncWrite};
+use tokio::task::JoinHandle;
+
+/// See [`AsyncTls::concurrent_handshakes`]
+#[cfg_attr(docsrs, doc(cfg(feature = "rt")))]
+#[derive(Clone, Debug)]
+pub struct ConcurrentHandshakeTls<T>(pub(crate) T);
```

Comment:
I think the original name may be a lot better, using "concurrent" here conflicts with the tokio docs:
https://tokio.rs/tokio/glossary

> Concurrency and parallelism are two related concepts that are both used when talking about performing multiple tasks at the same time. If something happens in parallel, then it also happens concurrently, but the opposite is not true: Alternating between two tasks, but never actually working on both at the same time is concurrency but not parallelism.

i.e. the base implementation is already concurrent, but using tokio spawn allows them to be parallel as well